### ISSUE TITLE
Improve way to incorporate third party objects

### ISSUE THREAD

**Issue by FlorianJacta** [Avaiga/taipy-gui#992]
**Description**
The goal would be to simplify how to incorporate third party objects like `fig` from Plotly. The relationship between `get_user_content_url`, `on_user_content` and the url can be complex for new comers or for complex applications using this to create multiple visualizations.

```python
from taipy.gui import Gui, get_user_content_url, notify


def expose_plotly(fig):
    buffer = io.StringIO()
    fig.write_html(buffer)
    return buffer.getvalue()


html = expose_plotly(sankey_graph(data))
uc_url = None


def on_user_content(state, path: str, query: dict):
    return state.html

def on_init(state):
    state.uc_url = get_user_content_url(state, "val", {"name": "param"})

```

**Acceptance Criteria**
- [ ] Ensure new code is unit tested, and check code coverage is at least 90%
- [ ] Propagate any change on the demos and run all of them to ensure there is no breaking change
- [ ] Ensure any change is well documented

**Comment by arcanaxion** [comment]
Indeed, beyond #980, I would also like the option of creating my own plotly figure object and having an easy way to pass it to taipy to display. I would still circumstantially want to use the taipy chart syntax (e.g. to use selected_traces), but I'm generally able to create more complicated charts quicker by using plotly.py itself
